libs/application/templates/parental-leave/src/fields/PeriodsRepeater/index.tsx

```diff
@@ -25,33 +41,116 @@ type FieldProps = {
 type ScreenProps = RepeaterProps & FieldProps
 
 const PeriodsRepeater: FC<ScreenProps> = ({
-  removeRepeaterItem,
   application,
   expandRepeater,
   field,
+  error,
+  setRepeaterItems,
+  setBeforeSubmitCallback,
+  setFieldLoadingState,
 }) => {
+  const [updateApplication] = useMutation(UPDATE_APPLICATION)
+  const editable =
+    application.state === States.DRAFT ||
+    application.state === States.EDIT_OR_ADD_PERIODS
+
   const showDescription = field?.props?.showDescription ?? true
   const dob = getExpectedDateOfBirth(application)
-  const { formatMessage } = useLocale()
-  const rights = getAvailablePersonalRightsInDays(application)
+  const { formatMessage, locale } = useLocale()
+  const rights = getAvailableRightsInDays(application)
   const daysAlreadyUsed = useDaysAlreadyUsed(application)
+  const { rawPeriods, periods } = getApplicationAnswers(application.answers)
+
+  useEffect(() => {
+    const syncPeriods = async () => {
+      setFieldLoadingState?.(true)
+      await setRepeaterItems(periods)
+      setFieldLoadingState?.(false)
+    }
+    // Upon entering this screen, if rawPeriods are not in sync with periods, sync them
+    // This means that there is an incomplete period inside answers.periods which we will remove
+    // TODO: real comparison
+    if (rawPeriods.length !== periods.length) {
+      syncPeriods()
+    }
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    if (!editable) {
+      return
+    }
+
+    if (rawPeriods.length === 0) {
+      expandRepeater()
+    }
+
+    setBeforeSubmitCallback?.(async () => {
+      try {
+        // Run answer validator on the periods selected by the user
+        const { errors } = await updateApplication({
+          variables: {
+            input: {
+              id: application.id,
+              answers: { validatedPeriods: periods },
+            },
+            locale,
+          },
+        })
+
+        if (errors) {
+          setFieldLoadingState?.(false)
```

Comment:
Does this callback need to change field loading state?

Reply:
No, this can be removed. Thanks.